src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java

```diff
@@ -437,7 +447,7 @@ && shouldStopProcessing(
           }
           // Add processed channels to the current blob, stop if we need to create a new blob
           blobData.add(channelsDataPerTable.subList(0, idx));
-          if (idx != channelsDataPerTable.size()) {
+          if (idx != channelsDataPerTable.size() || isIcebergMode) {
```

Comment:
I want to guard against the possibility of having isIcebergMode peppered across 10 different places in code, making it that much harder to maintain the client.

Ideally we should have one place in the whole client that does `if (isIcebergMode) { /* initialize booleans / ints / settings one way */ } else { /* initialize another way */ }` - it seems that the ParameterProvider is the best place to do this, or we introduce a separate ClientSettings class that holds all these settings that control different behaviors in different places in the client.

I also see there are three parameters on the parameter provider that need different values for iceberg - MAX_CHUNK_SIZE_IN_BYTES (set to 512 MB rn) and MAX_CHUNKS_IN_BLOB_AND_REGISTRATION_REQUEST_DEFAULT (set to 100 rn) and MAX_BLOB_SIZE_IN_BYTES (set to 1 GB rn). This is in addition to MAX_CLIENT_LAG needing an iceberg-specific value.

It looks like if parameterProvider.getMaxChunksInBlobAndRegistrationRequest() returns 1 then you'll get this same desired behavior ?

Note - in the parameterProvider you'll need to validate that customers can't override MAX_CHUNKS_IN_BLOB_AND_REGISTRATION_REQUEST and it can only ever be 1.

Reply:
Deleted the `isIcebergMode` param and use `MAX_CHUNKS_IN_BLOB_AND_REGISTRATION_REQUEST` for now.